### PR TITLE
Create static.yml

### PR DESCRIPTION
**Pull request by SocialNinjaMaster**
.github/workflows

### CODE REVIEW

File: .github/workflows/static.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# Simple workflow for deploying static content to GitHub Pages
+name: Deploy static content to Pages
```
Comment (SocialNinjaMaster) [inline .github/workflows/static.yml:2]: metadata.name
the pod's name
metadata.namespace
the pod's [namespace](https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces)
metadata.uid
the pod's unique ID
metadata.annotations['<KEY>']
the value of the pod's [annotation](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations) named <KEY> (for example, metadata.annotations['myannotation'])
metadata.labels['<KEY>']
the text value of the pod's [label](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels) named <KEY> (for example, metadata.labels['mylabel'])
The following information is available through environment variables but not as a downwardAPI volume fieldRef:

spec.serviceAccountName
the name of the pod's [service account](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/)
spec.nodeName
the name of the [node](https://kubernetes.io/docs/concepts/architecture/nodes/) where the Pod is executing
status.hostIP
the primary IP address of the node to which the Pod is assigned
status.podIP
the pod's primary IP address (usually, its IPv4 address)
The following information is available through a downwardAPI volume fieldRef, but not as environment variables:

metadata.labels
all of the pod's labels, formatted as label-key="escaped-label-value" with one label per line
metadata.annotations
all of the pod's annotations, formatted as annotation-key="escaped-annotation-value" with one annotation per line
Information available via resourceFieldRef
These container-level fields allow you to provide information about [requests and limits](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#requests-and-limits) for resources such as CPU and memory.

resource: limits.cpu
A container's CPU limit
resource: requests.cpu
A container's CPU request
resource: limits.memory
A container's memory limit
resource: requests.memory
A container's memory request
resource: limits.hugepages-*
A container's hugepages limit
resource: requests.hugepages-*
A container's hugepages request
resource: limits.ephemeral-storage
A container's ephemeral-storage limit
resource: requests.ephemeral-storage
A container's ephemeral-storage request

File: .github/workflows/static.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+# Simple workflow for deploying static content to GitHub Pages
```
Comment (SocialNinjaMaster) [inline .github/workflows/static.yml:1]: apiVersion: v1
kind: Pod
metadata:
  name: dapi-envars-fieldref
spec:
  containers:
    - name: test-container
      image: registry.k8s.io/busybox
      command: [ "sh", "-c"]
      args:
      - while true; do
          echo -en '\n';
          printenv MY_NODE_NAME MY_POD_NAME MY_POD_NAMESPACE;
          printenv MY_POD_IP MY_POD_SERVICE_ACCOUNT;
          sleep 10;
        done;
      env:
        - name: MY_NODE_NAME
          valueFrom:
            fieldRef:
              fieldPath: spec.nodeName
        - name: MY_POD_NAME
          valueFrom:
            fieldRef:
              fieldPath: metadata.name
        - name: MY_POD_NAMESPACE
          valueFrom:
            fieldRef:
              fieldPath: metadata.namespace
        - name: MY_POD_IP
          valueFrom:
            fieldRef:
              fieldPath: status.podIP
        - name: MY_POD_SERVICE_ACCOUNT
          valueFrom:
            fieldRef:
              fieldPath: spec.serviceAccountName
  restartPolicy: Never

Comment (SocialNinjaMaster) [reply .github/workflows/static.yml:1]: # If the new Pod isn't yet healthy, rerun this command a few times.
kubectl get pods

File: .github/workflows/static.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+on:
+  # Runs on pushes targeting the default branch
+  push:
+    branches: ["main"]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
```
Comment (SocialNinjaMaster) [inline .github/workflows/static.yml:12]: #

Comment (SocialNinjaMaster) [reply .github/workflows/static.yml:12]: minikube
dapi-envars-fieldref
default
172.17.0.4
default

**Review by SocialNinjaMaster — COMMENTED** [review]
# If the new Pod isn't yet healthy, rerun this command a few times.
kubectl get pods.
kubectl logs dapi-envars-fieldrefminikube
dapi-envars-fieldref
default

default

**Comment by Ninja-mann07** [comment]
```html
From 00a58b2bdab90d7665e886479ec5da4c41e6b70c Mon Sep 17 00:00:00 2001
From: SocialNinjaMaster <136135831+SocialNinjaMaster@users.noreply.github.com>
Date: Sat, 10 Jun 2023 01:05:33 -0700
Subject: [PATCH] Create static.yml

---
 .github/workflows/static.yml | 43 ++++++++++++++++++++++++++++++++++++
 1 file changed, 43 insertions(+)
 create mode 100644 .github/workflows/static.yml

diff --git a/.github/workflows/static.yml b/.github/workflows/static.yml
new file mode 100644
index 0000000..2bd90b5
--- /dev/null
+++ b/.github/workflows/static.yml
@@ -0,0 +1,43 @@
+# Simple workflow for deploying static content to GitHub Pages
+name: Deploy static content to Pages
+
+on:
+  # Runs on pushes targeting the default branch
+  push:
+    branches: ["main"]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+# Allow only one concurrent deployment, skipping runs queued between the run in-progress and latest queued.
+# However, do NOT cancel in-progress runs as we want to allow these production deployments to complete.
+concurrency:
+  group: "pages"
+  cancel-in-progress: false
+
+jobs:
+  # Single deploy job since we're just deploying
+  deploy:
+    environment:
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup Pages
+        uses: actions/configure-pages@v3
+      - name: Upload artifact
+        uses: actions/upload-pages-artifact@v1
+        with:
+          # Upload entire repository
+          path: '.'
+      - name: Deploy to GitHub Pages
+        id: deployment
+        uses: actions/deploy-pages@v2```

**Pull request merged**